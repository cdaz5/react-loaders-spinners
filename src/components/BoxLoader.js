import React from 'react';
import styled, { keyframes } from 'styled-components';
import { string, number, bool } from 'prop-types';

const fadeInOut = keyframes`
  0% {
    opacity: 0;
  }
  30% {
    opacity: 0;
  }
  90% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
`;

const Box = styled.div`
  position: relative;
  display: ${props => props.loading ? 'flex' : 'none'};
  width: ${props => props.width ? `${props.width}px` : '100px'};
  height: ${props => props.height ? `${props.height}px` : '100px'};

    div {
      width: 45%;
      height: 45%;
      background: ${props => props.pColor ? `${props.pColor}` : '#555'}
      opacity: 0;
      filter: blur(1px);
      animation: ${fadeInOut} 1s linear infinite;
    }

    div:nth-of-type(1) {
      position: absolute;
      top: 2.5%;
      left: 2.5%;
    }
    div:nth-of-type(2) {
      position: absolute;
      top: 2.5%;
      right: 2.5%;
      animation-delay: -0.25s;
    }
    div:nth-of-type(3) {
      position: absolute;
      bottom: 2.5%;
      right: 2.5%;
      animation-delay: -0.5s;
    }
    div:nth-of-type(4) {
      position: absolute;
      bottom: 2.5%;
      left: 2.5%;
      animation-delay: -0.75s;
    }
`;

const BoxLoader = props => (
  <Box {...props}>
    <div />
    <div />
    <div />
    <div />
  </Box>
);

BoxLoader.propTypes = {
  width: number,
  height: number,
  loading: bool.isRequired,
  pColor: string,
};

BoxLoader.defaultProps = {
  width: 100,
  height: 100,
  pColor: '#555',
  loading: true
};

export default BoxLoader;
