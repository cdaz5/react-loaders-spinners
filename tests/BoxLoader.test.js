import React from 'react';
import renderer from 'react-test-renderer';

import BoxLoader from '../src/components/BoxLoader';

describe('BoxLoader', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<BoxLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
