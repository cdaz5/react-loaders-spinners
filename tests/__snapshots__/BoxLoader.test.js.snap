// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxLoader matches snapshot 1`] = `
<div
  className="sc-bdVaJa gtRwnl"
  height={100}
  width={100}
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
